### ISSUE TITLE
Using Nginx as web server throws File[undef] error for Service[gunicorn]

### ISSUE THREAD

**Issue by rickatnight11** [echocat/puppet-graphite#289]
Using `dwerder-graphite v6.0.3` I have a simple manifest with:

```
class { '::graphite':
    gr_user                  => 'graphite',
    gr_group                 => 'graphite',
    gr_web_server            => 'nginx',
    gr_web_server_port       => '1000',
    gr_web_server_port_https => '1001',
    gr_line_receiver_port    => '2003',
    gr_disable_webapp_cache  => true,
    gr_storage_schemas       => [
      {
        name       => 'carbon',
        pattern    => '^carbon\.',
        retentions => '1m:90d',
      },
      {
        name       => 'default',
        pattern    => '.*',
        retentions => '1m:14d',
      },
    ],
  }
```

...that fails to apply the catalog with:

```
Error: Failed to apply catalog: Could not find dependency File[undef] for Service[gunicorn] at /etc/puppet/modules/graphite/manifests/config_gunicorn.pp:147
```

**Comment by dwerder** [comment]
Can you send more of the puppet log? Can you run the puppet in verbose? This could only happen, if the config.pp doesn't apply before the config_gunicorn.pp

**Comment by dwerder** [comment]
Can you try the develop branch? There where some fixes for gunicorn included.

**Comment by dwerder** [comment]
Just wait  a few minutes ;-) I will release version 6.0.4 ASAP

**Comment by rickatnight11** [comment]
Thanks!  This looks relevant:

https://github.com/echocat/puppet-graphite/commit/4ae73cc85cdbae5c7ba6407c5de4103ba00058fd#diff-f610cd45269ce3d0297cb21402d91588L146

:-)

**Comment by rickatnight11** [comment]
Confirmed fixed with 6.0.4 :-)  Thanks!
